src/handlers/showcase.ts

```diff
@@ -13,27 +13,49 @@ export default async function handleShowcaseMessage(
       config.FORTIES_SHOWCASE
     )) as TextChannel;
 
-    const url = msg.attachments.first()?.proxyURL as string
-    console.log(
-      `40s channel posted showcase: ${msg.author.username} ${url}`
-    );
+    const url = msg.attachments.first()?.proxyURL as string;
+
+    if (!url) {
+      console.log('Missing showcase image:', {
+        author: msg.author.tag,
+        message_url: msg.url,
+      });
+
+      await msg.reply(
+        "something went wrong. We'll take a look, but in the meantime please ensure an image is attached and try again."
+      );
+
+      return;
+    }
+
+    const trimDescription = (content: string): string => {
+      const trimmedContent = content
+        .replace(`<#${config.FORTIES_SHOWCASE}>`, '')
+        .trim();
+      return trimmedContent.length > 512
+        ? `${trimmedContent.substring(0, 512).trim()}...`
+        : trimmedContent;
+    };
```

Comment:
```suggestion
    const filterChannelName = (content: string, channel: string): string => {
      return content.replace(`<#${channel}>`, '').trim();
    };
    
    const truncateText = (content: string, characters: number = 512 ): string => {
      const formattedText = content.trim();
      return `${formattedText.substring(0, characters)}...`;
    };
```

What about something like this? That way each function is responsible for one thing and they could be reused elsewhere since we do similar things in multiple areas of the code. 

512 doesn't seem to be a hard limit so I guess it doesn't matter as much that we are technically returning 515 at max with the ellipsis, otherwise we can add a few operations in there. Unsure where 512 specifically came from and it seems a bit excessive IMO for a truncated message, but that's just my opinion.